spec/unit/jobs/services/service_usage_events_cleanup_spec.rb

```diff
@@ -5,6 +5,9 @@ module Jobs::Services
     RSpec.describe ServiceUsageEventsCleanup, job_context: :worker do
       let(:cutoff_age_in_days) { 30 }
       let(:logger) { double(Steno::Logger, info: nil) }
+      let!(:event_before_threshold) { ServiceUsageEvent.make(created_at: (cutoff_age_in_days + 1).days.ago) }
+      let!(:event_after_threshold) { ServiceUsageEvent.make(created_at: (cutoff_age_in_days - 1).days.ago) }
+      let!(:last_event_that_is_always_kept) { ServiceUsageEvent.make }
```

Comment:
Is `last_event_that_is_always_kept` intended to be used in a test in either this or the other spec?

Reply:
Hmm, no.

The tests even pass without the `last_event_that_is_always_kept` record, but you don't know if they're passing because `event_after_threshold` is actually after the threshold, or simply because it's the last record and we never delete that anymore.

Would it be more clear if I moved the event creation back into the individual tests? In that case `last_event_that_is_always_kept` is only really needed for the `keeps events created after the pruning threshold` test.

Or is there some other way to better show that intent?

I suppose we could add an explicit test in here to ensure that the last record is kept, although it feels a bit weird that all the edge cases are being tested at 3 different layers.

Reply:
I guess I don't really see any strong assertions around this behavior.  In all the specs I see, the assertions for the new tests are the same as the old tests, which is confusing.  Being very explicit about this unusual behavior would be good.

Reply:
Ok, I cleaned everything up a bit. Tried doing a bit more, but then other tests started to get weird, so it's a bit of a middle ground.